File: chapter2/chapter2_ws/src/demo_cpp_pkg/src/person_node.cpp
#include "rclcpp/rclcpp.hpp"
using namespace std;

class PersonNode : public rclcpp::Node // 继承rclcpp类别的Node节点
{
private:
    std::string name;
    int age;

public:
    PersonNode(const string &node_name, const string &name, const int &age)
        : Node(node_name) // 调用父类的构造函数
    // 类似于python中super().__init__(node_name)
    {
        this->name = name;
        this->age = age;
    };
    void eat(const string &food_name)
    {
        // cout<<"姓名："<<endl;
        RCLCPP_INFO(this->get_logger(), "我是%s,%d岁,爱吃%s", name.c_str(), age, food_name.c_str());
    }
};
int main(int argc,char ** argv)//程序入口参数
{
    // cout<<"参数变量"<<argc<<endl;
    // cout<<"程序名字"<<argv[0]<<endl;
    // string arg1=argv[1];
    // if(arg1=="--help")
    //     cout<<"error"<<endl;
    
    rclcpp::init(argc,argv);
    auto node=make_shared<PersonNode>("person_node","zhangsan",18);//创建节点
    RCLCPP_INFO(node->get_logger(),"你好C++节点");
    node->eat("鱼香肉丝");
    rclcpp::spin(node);
    rclcpp::shutdown();
    
    return 0;
}

File: chapter2/chapter2_ws/src/demo_python_pkg/demo_python_pkg/person_node.py
import rclpy
from rclpy.node import Node

class PersonNode(Node):
    def __init__(self,node_name:str,name:str,age:int)->None:
        super().__init__(node_name)
        self.name=name
        self.age=age
        pass
    def eat(self,food_name:str):
        """
        方法：吃东西
        :food_name：名字
        """
        # print(f"{self.name},{self.age}爱吃{food_name}")
        self.get_logger().info(f"{self.name},{self.age}爱吃{food_name}")#改为ros2节点

def main():
    rclpy.init()
    node=PersonNode("zhangsan","张三",18)
    node.eat("饭")
    rclpy.spin(node)
    rclpy.shutdown()